src/Runtime/python/PyExecutionSessionBase.cpp

```diff
@@ -54,6 +54,40 @@ struct npy_format_descriptor<onnx_mlir::float_16> {
 
 namespace onnx_mlir {
 
+//
+// Generate OMTensor's buffer for string input in pyArray
+//
+// For numerical array, pybind11 can convert multi-dimensional array into
+// one-dimensional array without manual conversion, but pybind11 cannot
+// convert multi-dimentional string array into one-dimentional array
+// automatically.
+// This function will be rewritten when pybind fixes the issue, or
+// better way for fixing it is found.
+//
+void *generateOMTensorBufferForStringData(py::array pyArray) {
+  auto shape = pyArray.shape();
+  uint64_t numElem = 1;
+  for (size_t i = 0; i < (size_t)pyArray.ndim(); ++i)
+    numElem *= shape[i];
+  uint64_t strLenTotal = 0;
+  uint64_t off = 0;
+  void *dataBuffer = NULL;
+  assert(pyArray.ndim() == 1 && "input pyArray should be flatten");
+  auto vec = pyArray.cast<std::vector<std::string>>();
+  for (int64_t i = 0; i < shape[0]; ++i)
+    strLenTotal += strlen(vec[i].data()) + 1;
+  dataBuffer = malloc(sizeof(char *) * numElem + strLenTotal);
+  assert(dataBuffer && "fail to alloc bufferData for string data");
```

Comment:
If possible when there are errors that are linked to the inputs, we try to have more friendly error messages, like throwing exceptions with a test message indicating the source of the error. Please do so if feasible.

Reply:
Updated the code to throw a python exception if memory allocation fails. Thanks.